impl/httpMetricsManager.go

```diff
@@ -11,49 +11,48 @@ import (
 type HttpMetricsManager struct {
 	requestSent int
 	countQueue  chan int
-	records     HttpRecords
-	durationSum DurationSummation
-	numErrors   int
-	// throughputRecord []int
+	// durationSum DurationSummation
+	numErrors  int
+	numRecords int
+	// records     HttpRecords
+	histogram map[time.Duration]int
 }
 
-// type HttpRecord struct {
-// 	duration time.Duration
-// 	id       uuid.UUID
-// 	status   string
-// }
-type HttpRecords []ChukonuHttpResponse
+// type HttpRecords []ChukonuHttpResponse
 
-type DurationSummation struct {
-	sum   time.Duration
-	index int
-}
+// type DurationSummation struct {
+// 	sum   time.Duration
+// 	index int
+// }
 
 // ------------sorting interface for struct HttpRecord-----------
-func (slice HttpRecords) Len() int {
-	return len(slice)
-}
-func (slice HttpRecords) Less(i, j int) bool {
-	return slice[i].duration < slice[j].duration
-}
-func (slice HttpRecords) Swap(i, j int) {
-	slice[i], slice[j] = slice[j], slice[i]
-}
+// func (slice HttpRecords) Len() int {
+// 	return len(slice)
+// }
+// func (slice HttpRecords) Less(i, j int) bool {
+// 	return slice[i].duration < slice[j].duration
+// }
+// func (slice HttpRecords) Swap(i, j int) {
+// 	slice[i], slice[j] = slice[j], slice[i]
+// }
 
 //----------------------------------------------------------------
 
 func (m *HttpMetricsManager) RecordRequest(request core.ChukonuRequest) {
 }
 func (m *HttpMetricsManager) RecordResponse(response core.ChukonuResponse) {
 	if response, ok := response.(ChukonuHttpResponse); ok {
-		m.records = append(m.records, response)
+		// m.records = append(m.records, response)
+		m.histogram[response.Duration()/time.Millisecond] += 1
+		m.numRecords += 1
```

Comment:
Good point, it's not thread safe indeed. Although no concurrency issue seems to happen during test. I will try to re implement this with channel later.